tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -478,6 +486,102 @@ def validate_synchronization(
+        """
+        fsid = self.get_fsid(cephfs_mirror_node)
+        filesystem_id = self.get_filesystem_id_by_name(source_clients, fs_name)
+        asok_file = self.get_asok_file(cephfs_mirror_node, fsid, daemon_name)
+        log.info("Validate the synced dir from cephfs-mirror status")
+        log.info("Install ceph-common on cephfs-mirror node")
+        cephfs_mirror_node.exec_command(
+            sudo=True, cmd="yum install -y ceph-common --nogpgcheck"
+        )
+        log.info("Get filesystem mirror status")
+        out, _ = cephfs_mirror_node.exec_command(
+            sudo=True,
+            cmd=f"cd /var/run/ceph/{fsid}/ ; ceph --admin-daemon {asok_file} "
+            f"fs mirror status {fs_name}@{filesystem_id} -f json",
+        )
+        data = json.loads(out)
+        log.info(data)
+        dir_count = data.get("snap_dirs", {}).get("dir_count")
+        if not data.get("snap_dirs"):
+            raise CommandFailed("Unable to get the Snap Dir")
+        if dir_count:
+            log.info(f"Snap Directory Count : {dir_count}")
+            log.info("Synchronization is successful")
+        else:
+            log.error("Snap Directory Count is not matching, Synchronization failed")
+            raise CommandFailed(
+                "Snap Directory Count is not matching, Synchronization failed"
+            )
+        log.info(dir_count)
+        return dir_count
+
+    @retry(CommandFailed, tries=5, delay=30)
+    def extract_synced_snapshots(
+        self,
+        cephfs_mirror_node,
+        fs_name,
+        fsid,
+        asok_file,
+        filesystem_id,
+        peer_uuid,
+    ):
+        """
+        Extracts the name of the last synced snapshot from the CephFS mirror status.
+
+        Args:
+        - cephfs_mirror_node: The node where the CephFS mirror is hosted
+        - fs_name: The name of the file system
+        - fsid: The FSID of the file system
+        - asok_file: The ASOK file for administrative commands
+        - filesystem_id: The ID of the filesystem
+        - peer_uuid: The UUID of the peer cluster
+
+        Returns:
+        - str: The name of the last synced snapshot, if found
+
+        Raises:
+        - CommandFailed: If the snapshot is not found or not synced after multiple attempts
+        """
+        log.info("Get peer mirror status")
+        log.info("Validate the snapshot sync to target cluster.")
+        cmd = (
+            f"cd /var/run/ceph/{fsid}/ ; ceph --admin-daemon {asok_file} fs mirror peer status "
+            f"{fs_name}@{filesystem_id} {peer_uuid} -f json"
+        )
+        out, _ = cephfs_mirror_node.exec_command(sudo=True, cmd=cmd)
+        data = json.loads(out)
+        for path, status in data.items():
+            last_synced_snap = status.get("last_synced_snap")
+            if last_synced_snap:
+                return last_synced_snap.get("name")
+
+        log.error("Snapshot not found or not synced")
```

Comment:
can we elaborate snapshot as "last synced snapshot not found or not synced"?

Reply:
fixed